pipeline/coaddition.py

```diff
@@ -424,3 +447,171 @@ def run(self, images):
             output.zogy_score = outscore
 
         return output
+
+
+class ParsCoaddPipeline(Parameters):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.date_range = self.add_par(
+            'date_range', 7.0, float, 'Number of days before end date to set start date, if start date is not given. '
+        )
+
+        self._enforce_no_new_attrs = True  # lock against new parameters
+
+        self.override(kwargs)
+
+
+class CoaddPipeline:
+    """A pipeline that runs coaddition and other tasks like source extraction on the coadd image. """
+    def __init__(self, **kwargs):
+        self.config = Config.get()
+
+        # top level parameters
+        self.pars = ParsCoaddPipeline(**(self.config.value('coaddition.pipeline', {})))
+        self.pars.augment(kwargs.get('pipeline', {}))
+
+        # coaddition process
+        coadd_config = self.config.value('coaddition.coaddition', {})
+        coadd_config.update(kwargs.get('coaddition', {}))
+        self.pars.add_defaults_to_dict(coadd_config)
+        self.coadder = Coadder(**coadd_config)
+
+        # source detection ("extraction" for the regular image!)
+        extraction_config = self.config.value('coaddition.extraction', {})
+        extraction_config.update(kwargs.get('extraction', {'measure_psf': True}))
+        self.pars.add_defaults_to_dict(extraction_config)
+        self.extractor = Detector(**extraction_config)
+
+        # astrometric fit using a first pass of sextractor and then astrometric fit to Gaia
+        astro_cal_config = self.config.value('coaddition.astro_cal', {})
```

Comment:
The comment I made above in the config yaml file.  You could put a line here that if astro_cal_config is (say) the string "default", then run astro_cal_config = self.config.value( 'astro_cal', {} )